examples/stein_dmm.py

```diff
@@ -0,0 +1,340 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Deep Markov Model inferred using SteinVI
+=================================================
+In this example we infer a deep Markov model (DMM) using SteinVI for generating music
+(chorales by Johan Sebastian Bach).
+
+The model DMM based on reference [1][2] and the
+[Pyro DMM example](https://github.com/pyro-ppl/pyro/blob/f73df6c1c20bc7b9164d79ce4217557d0aa8e396/examples/dmm/dmm.py).
+
+
+**Reference:**
+    1. Pathwise Derivatives for Multivariate Distributions
+    Martin Jankowiak and Theofanis Karaletsos (2019)
+    2. Structured Inference Networks for Nonlinear State Space Models [arXiv:1609.09869]
+    Rahul G. Krishnan, Uri Shalit and David Sontag (2016)
```

Comment:
I think this is a better link for Deep Markov Model: https://pyro.ai/examples/dmm.html , which can be put into references sections

Reply:
Makes sense. I'll update it.